### PR TITLE
KAFKA-6938: Add documentation for accessing Headers on Kafka Streams Processor API

### PR DESCRIPTION
**Pull request by jeqo**
Adding documentation to access Processor Context on Processor API.

### CODE REVIEW

**Comment by jeqo** [comment]
@guozhangwang here is a first draft of docs for accessing Processor Context. Let me now if the scope is fine. Also don't know how to test it. There is a gradle command to generate the docs?

File: docs/streams/developer-guide/processor-api.html [hunk 1]
```diff
@@ -53,6 +53,7 @@
                     </ul>
                 </li>
                 <li><a class="reference internal" href="#connecting-processors-and-state-stores" id="id8">Connecting Processors and State Stores</a></li>
+                <li><a class="reference internal" href="#accessing-processor-context" id="id10">Accessing Processor Context</a></li>
```
Comment (guozhangwang) [inline docs/streams/developer-guide/processor-api.html:56]: Why skip `id9` and use `id10` here?

File: docs/streams/developer-guide/processor-api.html [hunk 2]
```diff
@@ -351,6 +352,21 @@ <h2>
                     your store.</p>
             </div>
         </div>
+        <div class="section" id="accessing-processor-context">
+            <h2><a class="toc-backref" href="#id10">Accessing Processor Context</a><a class="headerlink" href="#accessing-processor-context" title="Permalink to this headline"></a></h2>
+            <p><code class="docutils literal"><span class="pre">ProcessorContext</span></code> gives access to metadata related with Processor state and the current record being processed.</p>
```
Comment (guozhangwang) [inline docs/streams/developer-guide/processor-api.html]: Add one more sentence at the beginning: `As we have mentioned in <href=#defining-a-stream-processor>Defining a Stream Processor<\href>, a <code>ProcessorContext</code> to control the processing workflow, such as scheduling a punctuation function, and committing the current processed state. This object can also be used to access the metadata related with...`

**Review by guozhangwang — COMMENTED** [review]
@jeqo you can follow this page https://cwiki.apache.org/confluence/display/KAFKA/Setup+Kafka+Website+on+Local+Apache+Server

to check the result of the added content (feel free to also add a snapshot of the rendered page in the PR for other reviewers to check).

File: docs/streams/developer-guide/processor-api.html (hunk 1 above)
Comment (jeqo) [reply docs/streams/developer-guide/processor-api.html:56]: `id9` is already used by: 

```
                <li><a class="reference internal" href="#unit-testing-processors" id="id9">Unit Testing Processors</a></li>
```

Comment (guozhangwang) [reply docs/streams/developer-guide/processor-api.html:56]: Ah I see, thanks.

**Comment by mjsax** [comment]
Should we add a section to the upgrade guide, "Streams API changes in 2.0.0" , too?

**Comment by guozhangwang** [comment]
@mjsax I had thought about that when reviewing. I think we do not necessarily need to record all API changes if they are not directly related to upgrade path.

**Comment by mjsax** [comment]
From my understanding, there are two sections: the one that describes relevant changes for upgrading only, and one that describes all API changes/improvements. Is my understanding incorrect?

**Comment by mjsax** [comment]
> From my understanding, there are two sections: the one that describes relevant changes for upgrading only, and one that describes all API changes/improvements. Is my understanding incorrect?

\cc @guozhangwang Comments?

**Comment by guozhangwang** [comment]
Sorry for the late reply.

I think you are correct. We should add this into the docs upgrade section. @jeqo could you file a follow-up PR? If you do not have time we can do it to make sure it gets in before the code freeze.

**Comment by jeqo** [comment]
Hello everyone,

Sorry but I do not have time these weeks (vacations). I'm back first week
of July.

**Comment by guozhangwang** [comment]
No worries, I can pick it up.

**Pull request merged**